android/examples/Example_Application/app/src/main/java/com/ironcorelabs/example_application/MainActivity.java

```diff
@@ -0,0 +1,140 @@
+package com.ironcorelabs.example_application;
+
+import android.app.Dialog;
+import android.content.DialogInterface;
+import android.os.Bundle;
+import android.text.Editable;
+import android.util.Log;
+import android.view.View;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AlertDialog;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.fragment.app.DialogFragment;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.ironcorelabs.sdk.DeviceContext;
+import com.ironcorelabs.sdk.DocumentDecryptResult;
+import com.ironcorelabs.sdk.DocumentEncryptOpts;
+import com.ironcorelabs.sdk.DocumentEncryptResult;
+import com.ironcorelabs.sdk.DocumentName;
+import com.ironcorelabs.sdk.GroupId;
+import com.ironcorelabs.sdk.IronOxide;
+import com.ironcorelabs.sdk.IronOxideConfig;
+import com.ironcorelabs.sdk.UserId;
+
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Objects;
+
+public class MainActivity extends AppCompatActivity implements MyRecyclerViewAdapter.ItemClickListener {
+    private EditText encryptText;
+    private EditText encryptId;
+    private TextView decryptView;
+    private IronOxide sdk;
+    private final ArrayList<DocumentEncryptResult> encryptedList = new ArrayList<>();
+    private MyRecyclerViewAdapter adapter;
+
+    private IronOxide initializeSdk() {
+        System.loadLibrary("ironoxide_android");
+        InputStream in = getResources().openRawResource(R.raw.device_context);
+        final java.util.Scanner s = new java.util.Scanner(in).useDelimiter("\\A");
+        final String deviceString = s.hasNext() ? s.next() : "";
+        try {
+            DeviceContext device = DeviceContext.fromJsonString(deviceString);
+            return IronOxide.initialize(device, new IronOxideConfig());
+        } catch (Exception e) {
+            e.printStackTrace();
+            Log.e("JNI", "Failed to load DeviceContext");
+            return null;
```

Comment:
Why return `null` here rather than just re-throw the Exception?

Reply:
I don't want to have `throws Exception` on the function because that makes it harder to call it from somewhere like onCreate. So I'm just logging the issue and moving on. Alternatively I could have it throw a RuntimeException and just crash the whole app, but that didn't seem great to me. Which do you prefer?

Reply:
Returning null is also going to crash their app with a NPE. Both are bad, but I'd argue that a RuntimeException lets them see the error soon and is therefore slightly better.

Reply:
At least from my testing on the emulator and my phone, the app did not crash even if it was given a blank device context. But I'm good with trying to make it more obvious as soon as there's an issue (especially a significant one like this), so I added the RuntimeException